internal/provider/resource_access_jaas_offer.go

```diff
@@ -0,0 +1,116 @@
+
+// Info implements the [resourceInfo] interface, used to extract the info from a Terraform plan/state.
+func (j offerInfo) Info(ctx context.Context, getter Getter, diag *diag.Diagnostics) (genericJAASAccessData, names.Tag) {
+	offerResource := jaasAccessOfferResourceOffer{}
+	diag.Append(getter.Get(ctx, &offerResource)...)
+	genericInfo := genericJAASAccessData{
+		ID:              offerResource.ID,
+		Users:           offerResource.Users,
+		Groups:          offerResource.Groups,
+		ServiceAccounts: offerResource.ServiceAccounts,
+		Access:          offerResource.Access,
+	}
+	// When importing, the offer url will be empty
+	var tag names.Tag
+	if offerResource.OfferUrl.ValueString() != "" {
+		// Create v4 tag but return v5 tag required by the interface.
+		// The v5 NewApplicationOfferTag expects a UUID.
+		tag = namesv4.NewApplicationOfferTag(offerResource.OfferUrl.ValueString())
+	}
+	return genericInfo, tag
+}
+
+// Save implements the [resourceInfo] interface, used to save info on Terraform's state.
+func (j offerInfo) Save(ctx context.Context, setter Setter, info genericJAASAccessData, tag names.Tag) diag.Diagnostics {
+	offerAccess := jaasAccessOfferResourceOffer{
+		OfferUrl:        basetypes.NewStringValue(tag.Id()),
+		ID:              info.ID,
+		Users:           info.Users,
+		Groups:          info.Groups,
+		ServiceAccounts: info.ServiceAccounts,
+		Access:          info.Access,
+	}
+	return setter.Set(ctx, offerAccess)
+}
+
+// ImportHint implements [resourceInfo] and provides a hint to users on the import string format.
+func (j offerInfo) ImportHint() string {
+	return "offer-<url>:<access-level>"
+}
+
+type jaasAccessOfferResource struct {
+	genericJAASAccessResource
+}
+
+type jaasAccessOfferResourceOffer struct {
+	OfferUrl        types.String `tfsdk:"offer_url"`
+	Users           types.Set    `tfsdk:"users"`
+	ServiceAccounts types.Set    `tfsdk:"service_accounts"`
+	Groups          types.Set    `tfsdk:"groups"`
+	Access          types.String `tfsdk:"access"`
+
+	// ID required for imports
+	ID types.String `tfsdk:"id"`
+}
+
+// Metadata returns metadata about the JAAS offer access resource.
+func (a *jaasAccessOfferResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_jaas_access_offer"
+}
+
+// Schema defines the schema for the JAAS offer access resource.
+func (a *jaasAccessOfferResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	attributes := a.partialAccessSchema()
+	attributes["offer_url"] = schema.StringAttribute{
+		Description: "The url of the offer for access management. If this is changed the resource will be deleted and a new resource will be created.",
+		Required:    true,
+		Validators:  []validator.String{
+			// No validator exists for offer URLs
+			// The names/v4 IsValidApplicationOffer is too restrictive
+			// and is not used by recent Juju 3 versions anyway.
```

Comment:
suggestion: You could write a wrapper around ParseOfferURL to validate the string.

https://github.com/juju/juju/blob/76a59def09b1acc73289ff7b420bc8432d48c9fe/core/crossmodel/offerurl.go#L74-L85

Reply:
Good idea, added